modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSource.java

```diff
@@ -621,9 +625,27 @@ protected void doInitialise() throws InitialisationException {
     flowConstruct = new LazyValue<>(() -> (FlowConstruct) componentLocator.find(getRootContainerLocation()).orElse(null));
     messageProcessContext = createProcessingContext();
     if (shouldRunOnThisNode()) {
+      if (LOGGER.isDebugEnabled()) {
+        boolean isPrimaryPollingInstance = clusterService.isPrimaryPollingInstance();
+        if (primaryNodeOnly) {
+          LOGGER
+              .debug("Message source '{}' on flow '{}' is being initialised, it is running on the primary node and it must only run on this node.",
+                     sourceModel.getName(), getLocation().getRootContainerName());
+        } else {
+          LOGGER
+              .debug("Message source '{}' on flow '{}' is being initialised, there are no cluster restrictions regarding which node must run the source. This node {} the primary node.",
```

Comment:
Message source ‘{}’ on flow ‘{}’ is initializing. Message source ‘{}’ is running on a non-primary node of the cluster that has primary node on node {}. Message source ‘{}’ can run on any node of the cluster.

Reply:
We actually don't have any info on the primary node, the original message says if this is the primary node or not. What about:

Message source ‘{}’ on flow ‘{}’ is initializing. Message source ‘{}’ is {} running on the primary node of the cluster. Message source ‘{}’ can run on any node of the cluster.

The placeholder added will either be "" or "not" so that it states if it is or is not the primary node.

Reply:
I would remove `Message source ‘{}’ can run on any node of the cluster.`. Logs should describe system events, not explain the product's functional definition.